### PR TITLE
#75 Automatically update releases on GitHub

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/ReleaseNotesPlugin.java [hunk 1]
```diff
@@ -51,46 +53,76 @@ public void apply(final Project project) {
...
+            task.setPreviewMode(previewMode);
+
+            LazyConfiguration.lazyConfiguration(task, () -> {
+                task.setGitHubWriteToken(conf.getGitHub().getWriteAuthToken());
+            });
+        });
+
+        updateReleaseNotesOnGitHubCleanupTask.setGitHubApiUrl(conf.getGitHub().getApiUrl());
+        updateReleaseNotesOnGitHubCleanupTask.setUpstreamRepositoryName(conf.getGitHub().getRepository());
+        updateReleaseNotesOnGitHubCleanupTask.setPublicationRepository("aa");
```
Comment (epeee) [inline subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/ReleaseNotesPlugin.java]: hmm, this one looks suspicious.

**Comment by epeee** [comment]
What about making new features like this one optional in the first place?
Additionally, what do you think about extracting the "release notes update to github" to a separate plugin?

File: subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/ReleaseNotesPlugin.java (hunk 1 above)
Comment (mstachniuk) [reply subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/ReleaseNotesPlugin.java]: Good catch! This line will be deleted by me in the next PR. The proper value is set in BintrayReleasePlugin.java:65

**Comment by mstachniuk** [comment]
In my opinion, there is no point to make it optional - a user doesn't need to configure anything, it's only added value. Also incubating fetures usually stay incubating very long (sometimes forever) what is not good for those features.

I thougt about separate plugin, it will be very simmilar to `ReleaseNotesPlugin` and needed to duplicate a lot of code. For compleatly new things, like https://github.com/mockito/shipkit/issues/233 I will create separate plugin. 

Do you think it should be separated? We can discuss on our Slack.
Thanks a lot for the review, better later then never :)

**Pull request merged**